eternalcore-core/src/main/java/com/eternalcode/core/feature/chat/ChatServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.eternalcode.core.feature.chat;
+
+import com.eternalcode.core.event.EventCaller;
+import com.eternalcode.core.feature.chat.event.restrict.ChatRestrictCause;
+import com.eternalcode.core.feature.chat.event.restrict.ChatRestrictEvent;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.injector.annotations.component.Service;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+
+@Service
+class ChatServiceImpl implements ChatService {
+
+    private final ChatSettings chatSettings;
+    private final Cache<UUID, Instant> slowdown;
+    private final EventCaller eventCaller;
+
+    @Inject
+    ChatServiceImpl(ChatSettings chatSettings, EventCaller eventCaller) {
+        this.chatSettings = chatSettings;
+        this.eventCaller = eventCaller;
+        this.slowdown = Caffeine.newBuilder()
+            .expireAfterWrite(this.chatSettings.getChatDelay().plus(Duration.ofSeconds(10)))
+            .build();
+    }
+
+    @Override
+    public void markUseChat(UUID userUuid) {
+        Duration chatDelay = this.chatSettings.getChatDelay();
+
+        ChatRestrictEvent event = new ChatRestrictEvent(userUuid, ChatRestrictCause.SLOWMODE);
```

Comment:
this fires ChatRestrictEvent, even though all above conditions from previous comment are filled. That does not sound logical to me. Maybe check slowmode != Duration.ZERO, then call event>?

Reply:
++